bioptim/dynamics/lagrange_interpolation.py

```diff
@@ -0,0 +1,252 @@
+from functools import cached_property
+
+import matplotlib.pyplot as plt
+import numpy as np
+from casadi import collocation_points, MX, SX
+
+
+class LagrangeInterpolation:
+    def __init__(
+        self,
+        time_grid: list[float],
+    ):
+        self.time_grid = time_grid
+
+    @cached_property
+    def polynomial_order(self):
+        return len(self.time_grid)
+
+    @classmethod
+    def from_grid_type(cls, polynomial_order: int, collocation_type: str = "radau"):
```

Comment:
is this the same as the variable "degree" ? If yes, could it be uniform throughout the code (no name preference) ?

Reply:
polynomial_order refers to Lagrange interpolation.

What would you think of polynomial_degree ?